### ISSUE TITLE
Task manager applications do not support the context menu in Wayland

### ISSUE THREAD

**Issue by ShalokShalom** [NixOS/nixpkgs#299190]
### Describe the bug

Hi there :)

I have a setup here on Plasma 6 unstable, that does not appear to support the right click context menu on Wayland for any of the task manager widgets anymore. 

It does not matter which of the alternatives (Icon only, Windows list, Task manager) and it does work fine in an X11 session. 

### Steps To Reproduce
1. https://paste.ee/p/0dSfh
2. https://paste.ee/p/neHcS

### Expected behavior
Right click menu to appear 

### Additional context
I also experience another issue, that might be related to it. 
Most of the times, when I rebuild, it renders the fixed (pinned) elements of the taskbar without logos and I need to restart Plasma, in order to fix this. 

This also results in me not being able to start applications anymore from the menu, but only from the command line. 

### Notify maintainers

@ilya-fedin
@K900
@LunNova
@mjm
@NickCao
@SuperSandro2000
@ttuegel

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.22, NixOS, 24.05 (Uakari), 24.05pre601756.44d0940ea560`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by K900** [comment]
Works fine here.

**Comment by ShalokShalom** [comment]
Yeah, also on my second laptop. 
Could it be hardware related?

**Comment by SuperSandro2000** [comment]
> Steps to reproduce the behavior:
> 
> 1. [paste.ee/p/0dSfh](https://paste.ee/p/0dSfh)
> 2. [paste.ee/p/neHcS](https://paste.ee/p/neHcS)

Can you please just paste the code in the issue? paste.ee is blocked by corp firewalls.

**Comment by SuperSandro2000** [comment]
> Could it be hardware related?

You probably have old config files laying around that cause issues.

**Comment by ShalokShalom** [comment]
```
# Edit this configuration file to define what should be installed on
# your system.  Help is available in the configuration.nix(5) man page
# and in the NixOS manual (accessible by running ‘nixos-help’).

{ config, pkgs, ... }:
let
  nix-software-center = import (pkgs.fetchFromGitHub {
    owner = "snowfallorg";
    repo = "nix-software-center";
    rev = "0.1.2";
    sha256 = "xiqF1mP8wFubdsAQ1BmfjzCgOD3YZf7EGWl9i69FTls=";
  }) {};
in
{
  imports =
    [ # Include the results of the hardware scan.
      ./hardware-configuration.nix
    ];

  # Bootloader.
  boot.loader.grub.enable = true;
  boot.loader.grub.device = "/dev/sda";
  boot.loader.grub.useOSProber = true;

  networking.hostName = "nixos"; # Define your hostname.
  # networking.wireless.enable = true;  # Enables wireless support via wpa_supplicant.

  # Configure network proxy if necessary
  # networking.proxy.default = "http://user:password@proxy:port/";
  # networking.proxy.noProxy = "127.0.0.1,localhost,internal.domain";

  # Enable networking
  networking.networkmanager.enable = true;

  # Set your time zone.
  time.timeZone = "Europe/Vienna";

  # Select internationalisation properties.
  i18n.defaultLocale = "en_US.UTF-8";

  i18n.extraLocaleSettings = {
    LC_ADDRESS = "de_AT.UTF-8";
    LC_IDENTIFICATION = "de_AT.UTF-8";
    LC_MEASUREMENT = "de_AT.UTF-8";
    LC_MONETARY = "de_AT.UTF-8";
    LC_NAME = "de_AT.UTF-8";
    LC_NUMERIC = "de_AT.UTF-8";
    LC_PAPER = "de_AT.UTF-8";
    LC_TELEPHONE = "de_AT.UTF-8";
    LC_TIME = "de_AT.UTF-8";
  };

  # Enable the X11 windowing system.
  services.xserver.enable = true;

  # Enable the KDE Plasma Desktop Environment.
  services.xserver.displayManager.sddm.wayland.enable = true;
  services.xserver.displayManager.sddm.enable = true;
  services.desktopManager.plasma6.enable = true;

  # Configure keymap in X11
  services.xserver = {
    xkb.layout = "at";
    xkb.variant = "nodeadkeys";
  };

  # Enable CUPS to print documents.
  services.printing.enable = true;

  # Enable sound with pipewire.
  sound.enable = true;
  hardware.pulseaudio.enable = false;
  security.rtkit.enable = true;
  services.pipewire = {
    enable = true;
    alsa.enable = true;
    alsa.support32Bit = true;
    pulse.enable = true;
    # If you want to use JACK applications, uncomment this
    #jack.enable = true;

    # use the example session manager (no others are packaged yet so this is enabled by default,
    # no need to redefine it in your config for now)
    #media-session.enable = true;
  };

  # Enable touchpad support (enabled default in most desktopManager).
  # services.xserver.libinput.enable = true;

  # Define a user account. Don't forget to set a password with ‘passwd’.
  users.users.cristof = {
    isNormalUser = true;
    description = "Cristof";
    extraGroups = [ "networkmanager" "wheel" ];
    packages = with pkgs; [
      audacity
      brave
      firefox
      gimp
      git
      hblock
      jamesdsp
      joplin
      kdePackages.dragon
      kdePackages.kcalc
      kdePackages.kdenlive
      kdePackages.yakuake
      libreoffice
      nix-software-center
      qbittorrent
      teamviewer
      telegram-desktop
      thunderbird
      signal-desktop
      zoom-us
    ];
  };

    programs.firefox = {
    enable = true;
    preferences = {
      "widget.use-xdg-desktop-portal.file-picker" = 1;
    };
  };

  services.teamviewer.enable = true;

  # Enable automatic login for the user.
  services.xserver.displayManager.autoLogin.enable = true;
  services.xserver.displayManager.autoLogin.user = "cristof";

  # Allow unfree packages
  nixpkgs.config.allowUnfree = true;

  # List packages installed in system profile. To search, run:
  # $ nix search wget
  environment.systemPackages = with pkgs; [
  ];

  # Some programs need SUID wrappers, can be configured further or are
  # started in user sessions.
  # programs.mtr.enable = true;
  # programs.gnupg.agent = {
  #   enable = true;
  #   enableSSHSupport = true;
  # };

  # List services that you want to enable:

  # Enable the OpenSSH daemon.
  # services.openssh.enable = true;

  # Open ports in the firewall.
  # networking.firewall.allowedTCPPorts = [ ... ];
  # networking.firewall.allowedUDPPorts = [ ... ];
  # Or disable the firewall altogether.
  # networking.firewall.enable = false;

  # This value determines the NixOS release from which the default
  # settings for stateful data, like file locations and database versions
  # on your system were taken. It‘s perfectly fine and recommended to leave
  # this value at the release version of the first install of this system.
  # Before changing this value read the documentation for this option
  # (e.g. man configuration.nix or on https://nixos.org/nixos/options.html).
  system.stateVersion = "24.05"; # Did you read the comment?

}
```
```
inxi -Fxz
System:    Kernel: 6.6.22 x86_64 bits: 64 compiler: gcc v: 13.2.0 Desktop: KDE Plasma 6.0.2 Distro: NixOS 24.05 (Uakari) 
Machine:   Type: Laptop System: LENOVO product: ThinkPad T440p v: 1.0 serial: <filter> 
           Mobo: LENOVO model: ThinkPad T440p v: 1.0 serial: <filter> BIOS: coreboot v: 4.19-293-gdb4b71ff10-dirty 
           date: 02/06/2023 
Battery:   ID-1: BAT0 charge: 45.1 Wh (100.0%) condition: 45.1/60.7 Wh (74.3%) volts: 12.2 min: 10.8 model: LGC 45N1147 
           status: Full 
CPU:       Info: Quad Core model: Intel Core i7-4800MQ bits: 64 type: MT MCP arch: Haswell rev: 3 cache: L2: 6 MiB 
           flags: avx avx2 lm nx pae sse sse2 sse3 sse4_1 sse4_2 ssse3 vmx bogomips: 43098 
           Speed: 2694 MHz min/max: 800/3700 MHz Core speeds (MHz): 1: 798 2: 800 3: 1802 4: 2483 5: 800 6: 800 7: 800 8: 800 
Graphics:  Message: No device data found. 
           Device-1: SunplusIT Integrated Camera type: USB driver: uvcvideo bus-ID: 3-12:7 
           Display: x11 server: X.org 1.21.1.11 driver: loaded: N/A resolution: <missing: xdpyinfo> 
           Message: Unable to show advanced data. Required tool glxinfo missing. 
Audio:     Device-1: HDA Intel HDMI driver: HDA-Intel message: bus/chip ids unavailable 
           Device-2: HDA Intel PCH driver: HDA-Intel message: bus/chip ids unavailable 
           Sound Server-1: ALSA v: k6.6.22 running: yes 
           Sound Server-2: PipeWire v: 1.0.4 running: yes 
Network:   Message: No device data found. 
           IF-ID-1: eno0 state: down mac: <filter> 
           IF-ID-2: wlp3s0 state: up mac: <filter> 
           IF-ID-3: wwp0s20u10i6 state: down mac: <filter> 
Bluetooth: Device-1: N/A type: USB driver: btusb v: 0.8 bus-ID: 3-11:6 
           Report: rfkill ID: hci0 rfk-id: 0 state: down bt-service: not found rfk-block: hardware: no software: no 
           address: see --recommends 
Drives:    Local Storage: total: 461.58 GiB used: 26.95 GiB (5.8%) 
           ID-1: /dev/sda vendor: Kingston model: SA400S37480G size: 447.13 GiB 
           ID-2: /dev/sdb type: USB vendor: Toshiba model: TransMemory size: 14.45 GiB 
Partition: ID-1: / size: 422.26 GiB used: 26.95 GiB (6.4%) fs: ext4 dev: /dev/sda1 
Swap:      ID-1: swap-1 type: partition size: 17.04 GiB used: 0 KiB (0.0%) dev: /dev/sda2 
Sensors:   Missing: Required tool sensors not installed. Check --recommends 
Info:      Processes: 213 Uptime: N/A Memory: 15.49 GiB used: 2.26 GiB (14.6%) Init: systemd Compilers: gcc: 13.2.0 
           Packages: 1211 Shell: Bash v: 5.2.26 inxi: 3.3.04 
```

**Comment by ShalokShalom** [comment]
> > Could it be hardware related?
> 
> You probably have old config files laying around that cause issues.

Considering that is the task manager (or rather all of them) and nothing else is effected, which config file would that be?
